spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/NettyRoutingFilter.java

```diff
@@ -124,8 +126,14 @@ public int getOrder() {
 
 					res.responseHeaders().forEach(entry -> headers.add(entry.getKey(), entry.getValue()));
 
-					if (headers.getContentType() != null) {
-						exchange.getAttributes().put(ORIGINAL_RESPONSE_CONTENT_TYPE_ATTR, headers.getContentType());
+					String contentTypeValue = headers.getFirst(HttpHeaders.CONTENT_TYPE);
+					if (StringUtils.hasLength(contentTypeValue)) {
+						try {
+							exchange.getAttributes().put(ORIGINAL_RESPONSE_CONTENT_TYPE_ATTR, headers.getContentType());
+						} catch (InvalidMediaTypeException e) {
+							//store non standard media types like "Content-Type: image"
+							exchange.getAttributes().put(ORIGINAL_RESPONSE_CONTENT_TYPE_ATTR, contentTypeValue);
```

Comment:
Wouldnt it just make sense to just call this line and not even bother calling `headers.getContentType()`?

Reply:
Let's just use the string version, and not deal with the other.

You'll also need to modify `ModifyResponseBodyGatewayFilterFactory` here this value is used.